src/cocotb/decorators.py

```diff
@@ -146,39 +190,50 @@ async def my_test(dut):
                 Passing a :class:`bool` value was removed.
                 Pass a specific :class:`Exception` or a tuple of Exceptions instead.
 
-        skip (bool, optional):
+        skip:
             Don't execute this test as part of the regression. Test can still be run
             manually by setting :make:var:`TESTCASE`.
 
-        stage (int)
+        stage:
             Order tests logically into stages, where multiple tests can share a stage.
             Defaults to 0.
+
+        name:
+            Override the default name of the test.
+            The default test name is the ``__qualname__`` of the decorated test function.
+
+            .. versionadded:: 2.0
     """
 
-    def wrapper(f: Union[Callable[..., None], TestFactory]) -> Optional[Test]:
+    if _func is not None:
+        cocotb.regression_manager.register_test(func=_func)
```

Comment:
Looks like when the test module is loaded outside sim environment (like when using Python runner) this fails because `cocotb.regression_manager` is `None`.

Reply:
I was never quite happy we allowed that. Oh well. Special case added.

Reply:
No happy about allowing what? The global being `None` ?

Reply:
Suggesting other tools read files with cocotb code in them.

Reply:
Ah. I would normally put them in separate modules, but even our tests include the pytest test and cocotb test in the same file.